tpm2_eventlog: illegal pointer cast

From a build on ARM:
```
lib/tpm2_eventlog_yaml.c: In function ‘yaml_uefi_var_data’:
lib/tpm2_eventlog_yaml.c:151:30: error: cast to pointer from integer of different size [-Werror=int-to-pointer-cast]
     uint8_t *variable_data = (uint8_t*)((uintptr_t)data->UnicodeName +
```

Also, an ugly piece of code in general... ;-)
```
    uint8_t *variable_data = (uint8_t*)((uintptr_t)data->UnicodeName +
                                        data->UnicodeNameLength *
                                        sizeof(char16_t));
```
Missing an `&` or somthing ?

--- a/lib/tpm2_eventlog_yaml.c
+++ b/lib/tpm2_eventlog_yaml.c
@@ -148,9 +148,8 @@ static bool yaml_uefi_var_data(UEFI_VARIABLE_DATA *data) {
     }
 
     char *var_data = calloc (1, VAR_DATA_HEX_SIZE(data));
-    uint8_t *variable_data = (uint8_t*)((uintptr_t)data->UnicodeName +
-                                        data->UnicodeNameLength *
-                                        sizeof(char16_t));
+    uint8_t *variable_data = (uint8_t*)&data->UnicodeName[
+        data->UnicodeNameLength];
     if (var_data == NULL) {
         LOG_ERR("failled to allocate data: %s\n", strerror(errno));
         return false;

